### DISCUSSION TITLE
Support for Open state styles for opened menu items in the Mega Menu

### DISCUSSION THREAD

**Discussion by Michaelgimii** [elementor/elementor#22426] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Hi @nicholaszein

ref : https://github.com/elementor/elementor/issues/22391

When hovering the megamenu item, its color changes (active mode). But if we place the mouse on the megamenu content, the color of the megamenu item returns to the default color. While the active color should still be displayed so that it is clear which menu item the megamenu content belongs to.

https://github.com/elementor/elementor/assets/88034902/9115cf81-f098-4236-84a0-8d88d5cfef17

### Describe the solution you'd like

Please add active mode support for mega menu items.  Just like what we see in the video below, from crocoblock mega manu.

https://github.com/elementor/elementor/assets/88034902/c679c6d9-a2bd-42a6-8342-776d8fc26c67

**Comment by nicholaszein** [comment]
Hi @Michaelgimii,

the **`Active`** state of the Mega Menu item, is related to the **Currently Active Page**, or **Currently Active Section** (for links with anchor points), and not to the _Currently Open Menu Item_. The color you see when you hover a menu item is the color you set in the **`Hover`** state. This is why when your mouse cursor leaves the item, the color returns to its **`Normal`** state.

If I understand correctly, what you are asking is a 4th state, where you define the styles for when a Menu Item with content is **`Open`**.

Is that correct?

**Reply by Michaelgimii** [reply]
>Is that correct?

yes, @nicholaszein

**Reply by nicholaszein** [reply]
Thanks for confirming @Michaelgimii.

I've updated the title of this feature request to reflect what you are asking for.

Kind regards

**Reply by forkanhossain** [reply]
Hi [nicholaszein](https://github.com/nicholaszein), Facing the same issue here. You should add this feature in the future update. Also Mega Menu click outside does not close. Need to update this feature too.

**Comment by idanishrangaiz** [comment]
Still facing htis issue also , the active state is not working i dont want to use any other plugin so i need to fix it with js or jquery or css , because elementor is not supporting the active state condition. i dont want it on hover state , i want it on active state

**Comment by devcomputer** [comment]
So it's been 2 years now and it is still not working, right?

**Comment by nordkite** [comment]
It is a shame how Elementor team doesn't release 100% needed features and doesn't even respond to this. I checked source code of Menu widget and there is no css active class added to the menu item.
